nixos/tests/systemd-networkd-dhcpserver.nix

```diff
@@ -98,12 +103,40 @@ import ./make-test-python.nix ({pkgs, ...}: {
         interfaces.eth1.useDHCP = true;
       };
     };
+
+    client2 = { config, pkgs, lib, ... }: {
+      virtualisation.vlans = [ 1 ];
+      systemd.services.systemd-networkd.environment.SYSTEMD_LOG_LEVEL = "debug";
+      networking = {
+        firewall.enable = false;
+        useNetworkd = true;
+        useDHCP = true;
+      };
+      # This is what we're testing here with client2, and as
+      # per this option's description this test doesn't make sense
+      # as this option is only meant to be used in the initrd, as
+      # `boot.initrd.systemd.network.useDHCP`.
+      #
+      # Unfortunately, testing the initrd here would be quite messy,
+      # so testing in stage2 will do:
```

Comment:
Could simply create an initrd service that's ordered after `network-online.target` that checks if the IP is configured and fails otherwise. Failure of services that `initrd.target` has a `Requires` dependency on should trigger emergency mode and cause a nixos test to fail.

Reply:
Oh, it seems we already have ~this in nixos/tests/systemd-initrd-networkd.nix. Going to move there.